src/V3EmitCHeaders.cpp

```diff
@@ -263,6 +265,45 @@ class EmitCHeader final : public EmitCConstInit {
             }
             puts("};\n}\n");
 
+            putns(sdtypep, "\nstd::vector<int> memberWidth(void) const {\n");
+            puts("return {");
+            for (const AstMemberDType* itemp = sdtypep->membersp(); itemp;
+                 itemp = VN_AS(itemp->nextp(), MemberDType)) {
+                if (itemp->isConstrainedRand()
+                    && (VN_IS(itemp->dtypep(), UnpackArrayDType)
+                        || VN_IS(itemp->dtypep(), DynArrayDType)
+                        || VN_IS(itemp->dtypep(), QueueDType)
+                        || VN_IS(itemp->dtypep(), AssocArrayDType))) {
+                    AstNodeDType* dtype = itemp->dtypep();
+                    // Traverse to the innermost sub-dtype to get the width
+                    while (dtype->subDTypep()) { dtype = dtype->subDTypep(); }
+                    putns(itemp, std::to_string(dtype->width()));
+                } else if (itemp->isConstrainedRand())
+                    putns(itemp, std::to_string(itemp->width()));
+                // Add comma if there is a next constrained random member
+                if (itemp->nextp() && VN_AS(itemp->nextp(), MemberDType)->isConstrainedRand())
+                    puts(",\n");
```

Comment:
Seems fragile, instead suggest just print at top of loop e.g.  

    if (needComma++) puts(",\n");
    
 And similar below